python/tests/res/enkf/test_log_config.py

```diff
@@ -50,39 +50,66 @@ def assert_log_config_load(
             write_abs_path=False
             ):
 
-
         with TestAreaContext("log_config_test") as work_area:
             work_area.copy_directory(self.case_directory)
 
+            config_dict = {}
             # Append config file
             with open(self.config_file, 'a') as cf:
                 if log_file:
+                    config_dict[ConfigKeys.LOG_FILE] = os.path.realpath(
+                        os.path.join(
+                            os.path.abspath(
+                                os.path.split(
+                                    self.config_file)[0]
+                            ), log_file
+                        )
+                    )
+
                     if write_abs_path:
-                        log_file = os.path.join(
-                                os.path.abspath(os.path.split(self.config_file)[0]),
-                                log_file
-                                )
+                        log_file = config_dict[ConfigKeys.LOG_FILE]
 
                     cf.write("\nLOG_FILE %s\n" % log_file)
 
+                else:
+                    config_dict[ConfigKeys.LOG_FILE] = os.path.realpath(
+                        os.path.join(
+                            os.path.abspath(
+                                os.path.split(
+                                    self.config_file)[0]
+                            ), "log.txt"
+                        )
+                    )
+
                 if log_level:
                     level = log_level
                     if sys.version_info[0] >= 3:
                         if not log_level.isalpha():
                             level = int(float(level))
                     cf.write("\nLOG_LEVEL %s\n" % level)
-
-            log_config = LogConfig(self.config_file)
+                    if type(level) is str and level.isdigit():
+                        config_dict[ConfigKeys.LOG_LEVEL] = MessageLevelEnum.to_enum(eval(level))
+                    elif type(level) is str:
+                        config_dict[ConfigKeys.LOG_LEVEL] = MessageLevelEnum.from_string("LOG_" + level)
+                    else:
+                        config_dict[ConfigKeys.LOG_LEVEL] = MessageLevelEnum.to_enum(level)
+                else:
+                    config_dict[ConfigKeys.LOG_LEVEL] = MessageLevelEnum.LOG_WARNING
+
+            log_config = LogConfig(user_config_file=self.config_file)
+            res_config = ResConfig(self.config_file)
+            log_config_dict = LogConfig(config_dict=config_dict)
+            self.assertTrue(log_config == log_config_dict)
+            self.assertTrue(log_config == res_config.log_config)
```

Comment:
I think here you could use `assertEqual`

Reply:
Assert equal will test of the pointers are identical, while == operator will utilize the __eq__ overloaded function